server/src/main/java/org/candlepin/dto/api/v1/PoolDTO.java

```diff
@@ -0,0 +1,1548 @@
+                this.productAttributes = new HashMap<String, String>();
+            }
+            else {
+                this.productAttributes.clear();
+            }
+            this.productAttributes.putAll(productAttributes);
+        }
+        else {
+            this.productAttributes = null;
+        }
+        return this;
+    }
+
+    /**
+     * Checks if the given product attribute has been defined on this poolDTO.
+     *
+     * @param key
+     *  The key (name) of the product attribute to lookup
+     *
+     * @throws IllegalArgumentException
+     *  if key is null
+     *
+     * @return
+     *  true if the product attribute is defined for this pool; false otherwise
+     */
+    @JsonIgnore
+    public boolean hasProductAttribute(String key) {
+        if (key == null) {
+            throw new IllegalArgumentException("key is null");
+        }
+
+        if (this.productAttributes == null) {
+            return false;
+        }
+
+        return this.productAttributes.containsKey(key);
+    }
+
+    /**
+     * Returns the identifier of stacked products/pools.
+     *
+     * @return the identifier of stacked products/pools.
+     */
+    @JsonProperty
+    public String getStackId() {
+        return stackId;
+    }
+
+    /**
+     * Sets the identifier of stacked products/pools.
+     *
+     * @param stackId set the identifier of stacked products/pools.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonIgnore
+    public PoolDTO setStackId(String stackId) {
+        this.stackId = stackId;
+        return this;
+    }
+
+    /**
+     * Returns true if this pool is stacked, false otherwise.
+     *
+     * @return true if this pool is stacked, false otherwise.
+     */
+    @JsonProperty
+    public Boolean isStacked() {
+        return stacked;
+    }
+
+    /**
+     * Sets if this pool is stacked or not.
+     *
+     * @param stacked set if this pool is stacked or not.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonIgnore
+    public PoolDTO setStacked(Boolean stacked) {
+        this.stacked = stacked;
+        return this;
+    }
+
+    /**
+     * Returns true if this is a development pool, false otherwise.
+     *
+     * @return true if this is a development pool, false otherwise.
+     */
+    @JsonProperty
+    public Boolean isDevelopmentPool() {
+        return developmentPool;
+    }
+
+    /**
+     * Sets if this is a development pool or not.
+     *
+     * @param developmentPool set if this is a development pool or not.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonIgnore
+    public PoolDTO setDevelopmentPool(Boolean developmentPool) {
+        this.developmentPool = developmentPool;
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @HateoasInclude
+    @Override
+    public String getHref() {
+        return this.id != null ? String.format("/pools/%s", this.id) : null;
```

Comment:
For methods like this which pull from other bits of data, since we're making it extensible, try to use the methods instead:
```    return this.getId() != null ? String.format(...) : null;```

That way subclasses need not override both methods should the source of the ID change.